### ISSUE TITLE
NPE when creating HSQLDB datasource without a username on the jdbc:ds-create  command

### ISSUE THREAD

**Issue by paulsp** [ops4j/org.ops4j.pax.jdbc#228]
Karaf 4.2.11
Pax-Jdbc-hsqldb 1.5.0

Creating a datasource using the command following command resulted in an NullPointerException, as listed below.
```
   jdbc:ds-create -url "jdbc:hsqldb:mem:." -dn "hsqldb" test-hsqldb
```

```
2021-03-23T16:34:46,000 | ERROR | CM Configuration Updater (Update: pid=org.ops4j.datasource.00b5088f-c420-47bd-9857-cd5357baee56) | configadmin                      | 11 - org.apache.felix.configadmin - 1.9.20 | [org.osgi.service.cm.ManagedServiceFactory, id=135, bundle=137&#x2F;mvn:org.ops4j.pax.jdbc&#x2F;pax-jdbc-config&#x2F;1.5.0]: Unexpected problem updating configuration org.ops4j.datasource.00b5088f-c420-47bd-9857-cd5357baee56
java.lang.NullPointerException: null
        at java.util.Hashtable.put(Hashtable.java:460) ~[?:1.8.0_172]
        at java.util.Properties.setProperty(Properties.java:166) ~[?:1.8.0_172]
        at org.hsqldb.jdbc.JDBCCommonDataSource.setUser(Unknown Source) ~[?:?]
        at org.ops4j.pax.jdbc.hsqldb.impl.HsqldbDataSourceFactory.setProperties(HsqldbDataSourceFactory.java:65) ~[?:?]
        at org.ops4j.pax.jdbc.hsqldb.impl.HsqldbDataSourceFactory.createDataSource(HsqldbDataSourceFactory.java:39) ~[?:?]
        at org.ops4j.pax.jdbc.config.impl.DataSourceRegistration.createDs(DataSourceRegistration.java:138) ~[?:?]
        at org.ops4j.pax.jdbc.config.impl.DataSourceRegistration.<init>(DataSourceRegistration.java:82) ~[?:?]
        at org.ops4j.pax.jdbc.config.impl.DataSourceConfigManager.lambda$null$4(DataSourceConfigManager.java:95) ~[?:?]
        at org.ops4j.pax.jdbc.config.impl.ServiceTrackerHelper$1.addingService(ServiceTrackerHelper.java:132) ~[?:?]
        at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:941) ~[osgi.core-6.0.0.jar:?]
        at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:870) ~[osgi.core-6.0.0.jar:?]
        at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256) ~[osgi.core-6.0.0.jar:?]
        at org.osgi.util.tracker.AbstractTracked.trackInitial(AbstractTracked.java:183) ~[osgi.core-6.0.0.jar:?]
        at org.osgi.util.tracker.ServiceTracker.open(ServiceTracker.java:318) ~[osgi.core-6.0.0.jar:?]
        at org.osgi.util.tracker.ServiceTracker.open(ServiceTracker.java:261) ~[osgi.core-6.0.0.jar:?]
        at org.ops4j.pax.jdbc.config.impl.ServiceTrackerHelper.track(ServiceTrackerHelper.java:141) ~[?:?]
        at org.ops4j.pax.jdbc.config.impl.DataSourceConfigManager.lambda$null$5(DataSourceConfigManager.java:91) ~[?:?]
        at org.ops4j.pax.jdbc.config.impl.ServiceTrackerHelper.track(ServiceTrackerHelper.java:147) ~[?:?]
        at org.ops4j.pax.jdbc.config.impl.ServiceTrackerHelper.track(ServiceTrackerHelper.java:86) ~[?:?]
        at org.ops4j.pax.jdbc.config.impl.DataSourceConfigManager.lambda$updated$6(DataSourceConfigManager.java:90) ~[?:?]
        at org.ops4j.pax.jdbc.config.impl.ServiceTrackerHelper.track(ServiceTrackerHelper.java:147) ~[?:?]
        at org.ops4j.pax.jdbc.config.impl.ServiceTrackerHelper.track(ServiceTrackerHelper.java:86) ~[?:?]
        at org.ops4j.pax.jdbc.config.impl.DataSourceConfigManager.updated(DataSourceConfigManager.java:89) ~[?:?]
        at org.apache.felix.cm.impl.helper.ManagedServiceFactoryTracker.updated(ManagedServiceFactoryTracker.java:159) ~[!/:?]
        at org.apache.felix.cm.impl.helper.ManagedServiceFactoryTracker.provideConfiguration(ManagedServiceFactoryTracker.java:93) [!/:?]
        at org.apache.felix.cm.impl.ConfigurationManager$UpdateConfiguration.run(ConfigurationManager.java:1416) [!/:?]
        at org.apache.felix.cm.impl.UpdateThread.run0(UpdateThread.java:122) [!/:?]
        at org.apache.felix.cm.impl.UpdateThread.run(UpdateThread.java:84) [!/:?]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_172]
```

**Comment by paulsp** [comment]
I submitted the pull request #229  to resolve this issue.

**Comment by grgrzybek** [comment]
Fixed here: 173128df344ef774b632425ba1201a7b7292ec90.

**Comment by grgrzybek** [comment]
Thank you @paulsp for the report and the PR!
